controllers/ccruntime_controller.go

```diff
@@ -228,6 +228,32 @@ func (r *CcRuntimeReconciler) processCcRuntimeDeleteRequest() (ctrl.Result, erro
 						return result, err
 					}
 					result, err = r.deleteUninstallDaemonsets()
+					prepostLabels := map[string]string{}
+					if r.ccRuntime.Spec.Config.PreInstall.Image != "" {
+						prepostLabels[PreInstallDoneLabel[0]] = PreInstallDoneLabel[1]
+					}
+					if r.ccRuntime.Spec.Config.PostUninstall.Image != "" {
+						prepostLabels[PostUninstallDoneLabel[0]] = PostUninstallDoneLabel[1]
+
+					}
+					err, nodes := r.getNodesWithLabels(prepostLabels)
+					if err != nil {
+						r.Log.Error(err, "an error occured when getting the list of nodes from which we want to"+
+							"remove preinstall/postuninstall labels")
+						return ctrl.Result{}, err
+					}
+					postuninstalledNodes := len(nodes.Items)
+					if !allNodesDone(postuninstalledNodes, r) {
+						return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 5}, err
```

Comment:
`RequeueAfter` is not required since for any errors, by default the request is queued with exponential backoff. It could be changed to `return ctrl.Result{}, err`

Reply:
good point Pradipta, this wasn't clear to me but it makes sense. I'll change it here and in other places

Reply:
I changed it, but left the `Requeue: true` in because in this case we definitely want to requeue but err is not set (allNodesDone doesn't set it, the value comes from a previous call which is already checked. 
So now it is `return ctrl.Result{Requeue: true}, nil`. 

I used your suggestion in the other case a bit further down. There are more places in the code that need cleanup but that's out of scope for this PR.